### DISCUSSION TITLE
Querying specific nodes

### DISCUSSION THREAD

**Discussion by beescuit** [ast-grep/ast-grep#584] [Q&A] [answered]
Is there any way to make the CLI return the content of a specific meta variable?

As an example: Doing security research, I want to list all endpoints of an api written in express.js. for this I would use something like this:

`app.$METHOD($ROUTE, $$$)`

Is it possible to make the output be a list of methods and/or routes? Would be very helpful for just "querying" code :)

**Comment by beescuit** [comment, marked as answer]
Managed to do this using jq: `sg -p 'app.$METHOD($ROUTE, $$$)' -l js --json | jq '[.[].metaVariables.single | {method: .METHOD.text, route: .ROUTE.text }]'`

Not sure if its the best way though 🤔

**Reply by HerringtonDarkholme** [reply]
Hi Rafel, thanks for giving ast-grep a try! I'm about to suggest the `--json` flag. 
The json output is designed for such use cases. 

This is a perfect solution from my perspective.
